src/ray/raylet/worker_pool.cc

```diff
@@ -1306,11 +1311,23 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
     int64_t num_needed = desired_usable_workers - num_usable_workers;
     RAY_LOG(DEBUG) << "Prestarting " << num_needed << " workers given task backlog size "
                    << backlog_size << " and available CPUs " << num_available_cpus;
-    for (int i = 0; i < num_needed; i++) {
-      PopWorkerStatus status;
-      StartWorkerProcess(
-          task_spec.GetLanguage(), rpc::WorkerType::WORKER, task_spec.JobId(), &status);
-    }
+    PrestartDefaultCpuWorkers(task_spec.GetLanguage(), num_needed);
+  }
+}
+
+void WorkerPool::PrestartDefaultCpuWorkers(ray::Language language, int64_t num_needed) {
+  static const WorkerCacheKey kDefaultCpuWorkerCacheKey{/*serialized_runtime_env*/ "",
+                                                        {{"CPU", 1}},
+                                                        /*is_actor*/ false,
+                                                        /*is_gpu*/ false};
```

Comment:
Thread safety footgun: `hash_` is mutable and can race if two threads invoke `Hash` concurrently. Recommend moving this away from `static` function-local.
https://github.com/ray-project/ray/blob/7ac449e5443c502e09141bb376cb8f24143c401f/src/ray/common/task/task_spec.h#L486